cookbook/agent_concepts/knowledge/filters/pdf_url/filtering.py

```diff
@@ -0,0 +1,91 @@
+"""
+User-Level Knowledge Filtering Example with PDF URLs
+
+This cookbook demonstrates how to use knowledge filters with PDF documents accessed via URLs,
+showing how to restrict knowledge base searches to specific cuisines, sources, or any other metadata attributes.
+
+Key concepts demonstrated:
+1. Loading PDF documents from URLs with specific metadata
+2. Filtering knowledge base searches by cuisine type
+3. Combining multiple filter criteria
+4. Comparing results across different filter combinations
+
+You can pass filters in the following ways:
+1. If you pass on Agent only, we use that for all runs
+2. If you pass on run/print_response only, we use that for that run
+3. If you pass on both, we override with the filters passed on run/print_response for that run
+"""
+
+from agno.agent import Agent
+from agno.knowledge.pdf_url import PDFUrlKnowledgeBase
+from agno.vectordb.qdrant import Qdrant
+
+# Set a unique collection name to avoid conflicts with other examples
+COLLECTION_NAME = "recipes-pdf-url"
+
+# Initialize the vector database
+vector_db = Qdrant(collection=COLLECTION_NAME, url="http://localhost:6333")
+
+# Step 1: Initialize knowledge base with URLs and metadata
+# ------------------------------------------------------------------------------
+# When initializing the knowledge base, we can attach metadata that will be used for filtering
+# This metadata can include cuisine type, source, region, or any other attributes
+
+knowledge_base = PDFUrlKnowledgeBase(
+    urls=[
+        {
+            "https://agno-public.s3.amazonaws.com/recipes/thai_recipes_short.pdf": {
+                "metadata": {
+                    "cuisine": "Thai",
+                    "source": "Thai Cookbook",
+                    "region": "Southeast Asia",
+                }
+            }
+        },
+        {
+            "https://agno-public.s3.amazonaws.com/recipes/cape_recipes_short_2.pdf": {
+                "metadata": {
+                    "cuisine": "Cape",
+                    "source": "Cape Cookbook",
+                    "region": "South Africa",
+                }
+            }
+        },
+    ],
+    vector_db=vector_db,
+)
+
+# Load all documents into the vector database
+knowledge_base.load(recreate=True)
```

Comment:
I would keep this `.load` method so it's possible to load more documents. But in these cookbooks it seems to be a needed step to load the documents presented on initialization. 

We could add another document here, to show the `.load` utility, or just remove the line so it doesn't look like a needed step?

Reply:
@manuhortet on the first run load is required mandatorily, else no documents will be added to the vector db. 
Maybe a comment above (# comment out after first run works?)

Reply:
I thought it wasn't mandatory - this looks OK then :)